sherpa/astro/xspec/__init__.py

```diff
@@ -217,6 +219,67 @@ def get_xsabundances() -> dict[str, float]:
             for name in get_xselements().keys()}
 
 
+# The current interface to the XSPEC model library makes this awkward
+# to write. Once the interface switches to using the FunctionUtility
+# interface this code will be a lot simpler iternally, without
+# changing the API.
+#
+def set_xsabundances(abundances: dict[str, float]) -> None:
+    """Set the abundances used by X-Spec.
+
+    .. versionadded:: 4.17.0
+
+    Parameters
+    ----------
+    abundances : dict
+        The new set of abundances. The keys are the element name
+        (e.g. 'Fe') and the values are the abundances. Any element
+        that is not present is set to 0.0, and other keywords are
+        ignored.
+
+    See Also
+    --------
+    get_xsabund, get_xsabundances, set_xsabund
+
+    Examples
+    --------
+
+    Change the relative abundance of Chromium:
+
+    >>> abunds = get_xsabundances()
+    >>> abunds['Cr'] = 4.2e-6
+    >>> set_xsabundances(abunds)
+
+    """
+
+    # Get the list of elemental values.
+    #
+    elems = get_xselements()
+    out = np.zeros(len(elems))
+    for name, abund in abundances.items():
+        # Skip unknown names. Should this warn the user?
```

Comment:
As discussed in other PRs, I feel that a better approach is to warn or error out. Typos do happen or people might not remember what elements are or are not in the XSPEC list. If my command returns without error or warning, then I would assume that it worked. 
This is new API and I can't see a case where it's beneficial to fail silently here.

Reply:
There's definitely an argument to be said here that a user who has set 'ar' to 2e-5 should have it pointed out that they should have used 'Ar'.

Now, we could make the keys case insensitive to fix this particular issue,  but we would still want to let a user know that we have ignored their Zirconium setting.

The advantage of warning (to the user): I can send in any-old dictionary and just have the elements I want processed.

The advantage of an error (to me): I can just error out at the first unknown value and so do not have to be concerned about generating many warning messages.

For this particular use case I think error-ing out is better - the user can just ensure that the input is a correctly-labelled dictionary. 

The one thing that would give me pause (ie prefer a warning) was if we thought the number of elements supported by XSPEC were to go up or down in the future. I don't see that likely...

Reply:
If the past is any guidance, XSPEC does almost never change the elements. And even then, I suspect that the most common user scenario is to get the dict using `avals=ui.get_xsabundances()`, modify a few numbers in `avals` and then set is back - and that will work even with a larger set of abundances. So let's make this an error, not a warning.